bluetooth/src/iosMain/kotlin/BluetoothMonitor.kt

```diff
@@ -19,42 +19,72 @@ package com.splendo.kaluga.bluetooth
 import com.splendo.kaluga.base.ServiceMonitor
+import com.splendo.kaluga.base.monitor.ServiceMonitorState
 import platform.CoreBluetooth.CBCentralManager
 import platform.CoreBluetooth.CBCentralManagerDelegateProtocol
+import platform.CoreBluetooth.CBCentralManagerStatePoweredOff
 import platform.CoreBluetooth.CBCentralManagerStatePoweredOn
+import platform.CoreBluetooth.CBManagerState
 import platform.darwin.NSObject
+import kotlin.coroutines.CoroutineContext
 
 actual interface BluetoothMonitor : ServiceMonitor {
+    /**
+     * Builder for [BluetoothMonitor].
+     * @param centralManager [CBCentralManager] used to set/unset a delegate and get info about bluetooth service status.
+     */
     actual class Builder(
         private val centralManager: CBCentralManager = CBCentralManager()
     ) {
-        actual fun create(): BluetoothMonitor = DefaultBluetoothMonitor(centralManager = centralManager)
+        /**
+         * Builder's create method.
+         * @param coroutineContext [CoroutineContext] used to define the coroutine context where code will run.
+         * @return [DefaultServiceMonitor]
+         */
+        actual fun create(coroutineContext: CoroutineContext): DefaultServiceMonitor =
+            DefaultBluetoothMonitor(
+                centralManager = centralManager,
+                coroutineContext = coroutineContext
+            )
     }
 }
 
 class DefaultBluetoothMonitor internal constructor(
-    private val centralManager: CBCentralManager
-) : DefaultServiceMonitor(), BluetoothMonitor {
+    private val centralManager: CBCentralManager,
+    coroutineContext: CoroutineContext
+) : DefaultServiceMonitor(coroutineContext), BluetoothMonitor {
 
     internal class CentralManagerDelegate(
-        private val updateEnabledState: () -> Unit
+        private val updateState: (CBManagerState) -> Unit
     ) : NSObject(), CBCentralManagerDelegateProtocol {
         override fun centralManagerDidUpdateState(central: CBCentralManager) {
-            updateEnabledState()
+            updateState(central.state)
         }
     }
 
     private val centralManagerDelegate = CentralManagerDelegate(::updateState)
-    override val isServiceEnabled: Boolean
-        get() = centralManager.state == CBCentralManagerStatePoweredOn
 
     override fun startMonitoring() {
         super.startMonitoring()
         centralManager.delegate = centralManagerDelegate
+
+        updateState(centralManager.state)
     }
 
     override fun stopMonitoring() {
         super.stopMonitoring()
         centralManager.delegate = null
     }
+
+    private fun updateState(status: CBManagerState) {
+        launchTakeAndChangeState {
+            {
+                when (status) {
+                    CBCentralManagerStatePoweredOn -> ServiceMonitorState.Initialized.Enabled
+                    CBCentralManagerStatePoweredOff -> ServiceMonitorState.Initialized.Disabled
+                    else -> ServiceMonitorState.NotSupported
```

Comment:
In case of `CBManagerStateUnauthorized` service technically supported, but permissions not granted.
In case of `CBManagerStateResetting` service also supported, but not determined.

I think we need to emit `NotSupported` when it's really not supported? Open question

Reply:
You are right, is not correct to return that in cases where actually is supported, but.. I apply changes to fix this. At this point I should also integrate kaluga permissions in it to cover not granted permissions

Reply:
If it's powered off, it's disabled (I hope). If it's not supported/unknown — it's not present (simulator).
In other cases it's enabled (Bluetooth ON in system settings)